src/util/jsPlumbHelpers.js

```diff
@@ -26,48 +29,46 @@ export const connectLinkItems = (
     PaintStyle: { strokeWidth: 1, stroke: '#000000', dashstyle: '4 2' },
     Anchors: ['Left', 'Left'],
     deleteEndpointsOnDetach: false,
-    Overlays: [
-      [
-        'Custom',
-        {
-          create: component => {
-            if (!component.source) return container.starButton.current;
-            const { isPrimary } = connectionArray.find(
-              connection => connection.targetId === parent,
-            );
-            return isPrimary
-              ? container.starButton.current
-              : container[`linkButton-${subject}`];
-          },
-          location: 57,
-          id: 'sourceOverlay',
-        },
-      ],
-    ],
+    Overlays: isPrimary
+      ? [
+          [
+            'Custom',
+            {
+              create: component => {
+                return container.starButton.current;
+              },
+              location: 57,
+              id: 'sourceOverlay',
+            },
+          ],
+        ]
+      : [],
   });
 
+  if (!document.getElementById(source)) {
+    // Sometimes the menu item is not rendered when showLink is called
+    await new Promise(resolve => setTimeout(resolve, 500));
```

Comment:
Ser ut som en liten fin hack 😅

Reply:
Haha, ja. I stedet for å lage masse logikk med en callback fra child component som sa fra når den var ferdig rendet, så syns jeg dette er godt nok, har ingenting å si at det er en liten delay på visning av link, og rendering av meny tar aldri lengre tid enn 500ms